src/main/java/org/craftercms/studio/impl/v2/repository/GitContentRepository.java

```diff
@@ -964,8 +964,12 @@ public void publish(String site, String sandboxBranch, List<DeploymentItemTO> de
                         commitId = deploymentItem.getCommitId();
                         path = helper.getGitPath(deploymentItem.getPath());
                         if (Objects.isNull(commitId)) {
-                            logger.warn("Skipping file " + path + " because commit id is null");
-                            continue;
+                            if (contentExists(site, path)) {
+                                commitId = getRepoLastCommitId(site);
+                            } else {
+                                logger.warn("Skipping file " + path + " because commit id is null");
```

Comment:
If the file doesn't exist on disk, it means it exists only in the database. Should this trigger removal from the database (including workflow/etc.)?

Reply:
we agreed that it should be skipped. no other action